src/NzbDrone.Core/Messaging/Commands/CommandExecutor.cs

```diff
@@ -1,117 +1,65 @@
 using System;
-using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using NLog;
 using NzbDrone.Common;
-using NzbDrone.Common.EnsureThat;
-using NzbDrone.Common.Serializer;
 using NzbDrone.Common.TPL;
-using NzbDrone.Core.Messaging.Commands.Tracking;
+using NzbDrone.Core.Lifecycle;
 using NzbDrone.Core.Messaging.Events;
 using NzbDrone.Core.ProgressMessaging;
 
 namespace NzbDrone.Core.Messaging.Commands
 {
-    public class CommandExecutor : ICommandExecutor
+    public class CommandExecutor : //IHandle<CommandQueuedEvent>,
+                                   IHandle<ApplicationStartedEvent>,
+                                   IHandle<ApplicationShutdownRequested>
     {
         private readonly Logger _logger;
         private readonly IServiceFactory _serviceFactory;
-        private readonly ITrackCommands _trackCommands;
+        private readonly ICommandService _commandService;
         private readonly IEventAggregator _eventAggregator;
         private readonly TaskFactory _taskFactory;
 
-        public CommandExecutor(Logger logger, IServiceFactory serviceFactory, ITrackCommands trackCommands, IEventAggregator eventAggregator)
+        private static CancellationTokenSource _cancellationTokenSource;
+        private const int THREAD_LIMIT = 1;
```

Comment:
I think we can leave it for future. also, 3 threads working on disk isn't that bad.